api/mfa/mfa.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mfa
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"strings"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gravitational/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/gravitational/teleport/api/client/proto"
+)
+
+const (
+	mfaResponseToken                 = "mfa_challenge_response"
+	errAdminActionMFARequiredMessage = "admin-level API request requires MFA verification"
+)
+
+// NewErrAdminActionMFARequired returns an error indicating that an admin-level
+// API request failed due to missing or failing MFA verification.
```

Comment:
> due to missing or failing MFA verification

Could we use the error just for missing MFA? Failed MFA likely shouldn't be retried automatically. (Or do we want that?)